juniper/src/executor/mod.rs

```diff
@@ -743,6 +746,55 @@ where
             })
     }
 
+    /// Construct a lookahead selection for the current selection.
+    ///
+    /// This allows seeing the whole selection and perform operations
+    /// affecting the children.
+    pub fn look_ahead_lazy(&'a self) -> look_ahead_lazy::LookAheadSelection<'a, S> {
```

Comment:
The replacement for `Executor::look_ahead`. 

As far as I can see, there are no tests for `Executor::look_ahead`. So I don't know whether the `FragmentSpread` fallback branch at the bottom actually works or not. The first branch where it finds the field works, I have tested this manually.

Reply:
@audunhalland 

> edit: I found the tests, there are a lot! But testing just executor.look_ahead(); after it turns lazy is arguably a bit weaker. I suggest at least adding a level of .children(); to the integration tests.

Will be happy to see this 👍

Reply:
I will fix all the tests after (if!) we reach agreement on an API and the old look_ahead has been removed.